var/spack/repos/builtin/packages/meshkit/package.py

```diff
@@ -0,0 +1,71 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+#
+from spack import *
+
+
+class Meshkit(AutotoolsPackage):
+    """MeshKit is an open-source library of mesh generation functionality.
+       Its design philosophy is two-fold: it provides a collection of 
+       meshing algorithms for use in real meshing problems, along with
+       other tools commonly needed to support mesh generation"""
+
+    homepage = "http://sigma.mcs.anl.gov/meshkit-library"
+    url = "http://ftp.mcs.anl.gov/pub/fathom/meshkit-1.5.0.tar.gz"
+
+    version('1.5.0',       '90b52416598ef65525ce4457a50ffe68')
+    version('1.4.1',       'a8cee0babf32be0f3964a5a2a99f7e5b')
+    version('1.4.0',       '5d7799ee528045b20e78e5c900006e92')
+    version('1.3-nightly', 'f56129ce1f3beae83a4038b47ae6b2c2')
+    version('1.3',         '19bf51f56392f28d70e581fd4c2f51c8')
+
+    variant("mpi", default=True, description='enable mpi support')
+    variant("debug", default=False, description='enable debug symbols')
+    variant("shared", default=False, description='enable shared builds')
+    
+    depends_on('mpi', when='+mpi')
+    depends_on('cgm')
+    depends_on('moab+irel+fbigeom')
+
+    def configure_args(self):
+        spec = self.spec
+        args = [ 
+            "--with-igeom={0}".format(spec['cgm'].prefix),
+            "--with-imesh={0}".format(spec['moab'].prefix)
+        ]
+        if '+mpi' in spec:
+            args.extend([
+                "--with-mpi",
+                "CC={0}".format(spec['mpi'].mpicc),
+                "CXX={0}".format(spec['mpi'].mpicxx),
+                "FC={0}".format(spec['mpi'].mpifc)
+            ])
+
+        if '+debug' in spec:
+            args.append("--enable-debug")
+        if '+shared' in spec:
+            args.append("--enable-shared")
```

Comment:
is that a policy? what about default arguments that are turned on / off? should we always be very explicit by turning them on / off ?